src/main/kotlin/me/zeroeightsix/kami/feature/module/Nuker.kt

```diff
@@ -0,0 +1,126 @@
+package me.zeroeightsix.kami.feature.module
+
+import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting
+import me.zero.alpine.listener.EventHandler
+import me.zero.alpine.listener.Listener
+import me.zeroeightsix.kami.event.TickEvent
+import me.zeroeightsix.kami.kotlin
+import me.zeroeightsix.kami.util.asVec3d
+import me.zeroeightsix.kami.util.singleVec
+import net.minecraft.block.FluidBlock
+import net.minecraft.client.network.ClientPlayerEntity
+import net.minecraft.client.world.ClientWorld
+import net.minecraft.network.packet.c2s.play.PlayerActionC2SPacket
+import net.minecraft.util.math.BlockPos
+import net.minecraft.util.math.Box
+import net.minecraft.util.math.Direction
+import net.minecraft.util.math.Vec3d
+import net.minecraft.world.RaycastContext
+
+@Module.Info(
+    name = "Nuker",
+    category = Module.Category.PLAYER,
+    description = "Destroys Blocks around the Player"
+)
+object Nuker : Module() {
+    @Setting
+    private var hitRange: @Setting.Constrain.Range(min = 0.0, max = 8.0, step = 1.0) Double = 4.0
+
+    @Setting
+    private var throughBlocks = false
+
+    @Setting(comment = "Select the best tool for the block being broken")
+    private var selectTool = false
+
+    @Setting(comment = "Only break blocks that can be instantly broken")
+    private var onlyInstant = false
+
+    private var progress = 0.0
+    private var currentBlock: BlockPos? = null
+
+    @EventHandler
+    private val updateListener = Listener<TickEvent.InGame>({
+        instantMineBlocks(it.player, it.world)
+
+        if (currentBlock == null || validate(it.player, it.world, currentBlock!!))
+            currentBlock = nextBlock(it.player, it.world)
+
+        if (!it.player.isCreative && !onlyInstant) {
+            val block = currentBlock ?: return@Listener
+            val state = it.world.getBlockState(block)
+
+            if (selectTool)
+                AutoTool.equipBestTool(state)
+
+            if (progress == 0.0)
+                mine(block, true)
+
+            progress += state.calcBlockBreakingDelta(it.player, it.world, block)
+            if (progress >= 1.0) {
+                mine(block, false)
+                progress = 0.0
+                currentBlock = null
+            }
+        }
+    })
+
+    private fun getBoxCorner(playerPos: Vec3d, negative: Boolean = false) =
+        playerPos.add(Vec3d(hitRange, hitRange, hitRange).run { if (negative) negate() else this })
+
+    private fun getValidBlocks(player: ClientPlayerEntity, world: ClientWorld) =
+        BlockPos.method_29715(Box(getBoxCorner(player.pos), getBoxCorner(player.pos, true)))
+            .filter { validate(player, world, it) }
+
+    /**
+     * mines a block using packets
+     * @param op if true, mining is started, if false it is stopped
+     */
```

Comment:
```suggestion
    /**
     * Mines a block using packets
     * 
     * @param op if `true`, mining is started. Otherwise, it is stopped.
     */
```